plugins/inputs/win_perf_counters/win_perf_counters.go

```diff
@@ -148,34 +155,69 @@ func (m *Win_PerfCounters) SampleConfig() string {
 	return sampleConfig
 }
 
-func (m *Win_PerfCounters) ParseConfig(metrics *itemList) error {
-	var query string
+func expandCounterQuery(query string) ([]string, error) {
+	var bufSize uint32
+	var buf []uint16
+	ret := PdhExpandWildCardPath(query, nil, &bufSize)
+	for ret == PDH_MORE_DATA {
+		buf = make([]uint16, bufSize)
```

Comment:
This code does not seem defensive enough to me.  Can bufSize be zero here, if so then the next line will panic.  What if ret is not PDH_MORE_DATA at least once,  then we will try to index it in UTF16ToStringArray and panic.  In general we need to check anything that crosses an application boundary before we can assume anything about it.